### DISCUSSION TITLE
Don't remove / bring back Long (64-bit integer) type

### DISCUSSION THREAD

**Discussion by aleksandarsusnjar** [graphql-java/graphql-java#2344] [Ideas]
In 16.x this scalar type was deprecated. I see that in present `master` it has been removed. Sure, old JavaScript and JSON do not have a way to represent it but:

1. JavaScript isn't the only language GraphQL clients use.
2. Current JavaScript can express large integers now. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt
3. `Long`/`long` are very commonly used types in Java, for various purposes.
4. Custom scalars are already possible/allowed. Hence, a `graphql-java`-based server developer gets to decide whether or not they are OK with any particular type, for any reason and whether they can, want to or are otherwise content with forcing the clients to support those types. `Long` isn't special in that sense other than it was already supported and is common in the ecosystem `graphql-java` targets. For the same reason I don't think it should require an extension library ... and on top of that, this is a primitive Java type we're talking about.

Also note that JSON / ECMA-404 standard does **NOT** prevent 64-bit integers. See  https://www.ecma-international.org/publications-and-standards/standards/ecma-404/ 

> **8 Numbers**
>
> A number is a sequence of decimal digits with no superfluous leading zero. It may have a preceding minus
sign (U+002D). It may have a fractional part prefixed by a decimal point (U+002E). It may have an exponent,
prefixed by e (U+0065) or E (U+0045) and optionally + (U+002B) or – (U+002D). The digits are the code
points U+0030 through U+0039.
>
> _Figure 4 - number_ 
>
> Numeric values that cannot be represented as sequences of digits (such as Infinity and NaN) are not
permitted.

Figure 4 shows a syntax diagram that allows for any size integer. No restrictions are included in the description or the fine print.

Finally, because GraphQL defines a strict schema, there is no ambiguity as to what kind of number a 64-bit `Long` would be. BTW, I would apply the same thinking to `Byte`/`byte` and `Short`/`short` as well - it is up the schema/server/API designer to decide whether or not to use those types.

Also see:

- https://www.npmjs.com/package/apollo-type-bigint
- https://www.npmjs.com/package/json-bigint
- https://graphql-ruby.org/api-doc/1.9.17/GraphQL/Types/BigInt.html
- https://dgraph.io/docs/graphql/schema/types/

**Comment by andimarek** [comment]
Long is still available as part of https://github.com/graphql-java/graphql-java-extended-scalars .

We made the decision to not include non standard Scalars in the base library to keep it focused. Therefore there is nothing which prevents you in using Long.
